src/main/java/com/yelp/nrtsearch/server/nrt/NRTReplicaNode.java

```diff
@@ -194,13 +194,7 @@ private static CopyState readCopyState(com.yelp.nrtsearch.server.grpc.CopyState
     long version = copyState.getVersion();
     Map<String, FileMetaData> files = readFilesMetaData(copyState.getFilesMetadata());
 
-    int count = copyState.getCompletedMergeFilesCount();
-    assert count == copyState.getCompletedMergeFilesCount();
-
-    Set<String> completedMergeFiles = new HashSet<>();
-    for (String completedMergeFile : copyState.getCompletedMergeFilesList()) {
-      completedMergeFiles.add(completedMergeFile);
-    }
+    Set<String> completedMergeFiles = new HashSet<>(copyState.getCompletedMergeFilesList());
```

Comment:
Some of these are pretty good, like this one might be better performant too (if the JIT was not optimizing it already).